fastqheat/__main__.py

```diff
@@ -74,23 +89,37 @@ def download_run_ftp(accession, output_directory, **kwargs):
     bool
         True if run was correctly downloaded, otherwise - False
     """
-    correctness = []
     ftps, md5s = metadata.get_urls_and_md5s(accession)
+    successful = True
+    accession_directory = Path(output_directory, accession)
+    accession_directory.mkdir(parents=True, exist_ok=True)
 
     for ftp, md5 in zip(ftps, md5s):
         srr = ftp.split('/')[-1]
-        bash_command = f'mkdir -p {output_directory}/{accession} && curl -L {ftp} -o {output_directory}/{accession}/{srr}'  # noqa: E501 line too long - will be fixed in next PRs
-        logging.debug(bash_command)
-        logging.info('Try to download %s file', srr)
-        # execute command in commandline
-        os.system(bash_command)
-        # check completeness of the file and return boolean
-        correctness.append(md5_checksum(srr, f"{output_directory}/{term}", md5))
-
-    if all(correctness):
+        logging.info('Trying to download %s file', srr)
+        file_path = accession_directory / srr
+
+        _download_file(_force_ena_http(ftp), file_path)
```

Comment:
I respect your desire to make the code clearer, but is one line of code worth a separate function?

What do you think about actually changing the function that returns this list of links and force the schema there?
For example:
```python
def get_urls_and_md5s(term: str, ftp=False, aspera=False) -> tuple[list[str], list[str]]:
    ...
    if ftp:
        # FTP URLs from ENA do NOT currently include the scheme. Just prepend http://
        # https://ena-docs.readthedocs.io/en/latest/retrieval/file-download.html
        urls = [f"http://{uri}" for uri in response.json()[0][url_type].split(';')]
    else:
        urls = response.json()[0][url_type].split(';')

    ...
```

Reply:
> I respect your desire to make the code clearer, but is one line of code worth a separate function?

IMO yes, because subjectively it looks cleaner this way, because it also contains comments, and because it'll be easier to change if ENA API starts returning URLs with scheme. I made the change you requested though.

Reply:
Thanks!